modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1437,6 +1450,49 @@ private CompletableFuture<Void> finishAndCleanup(
             UUID txId,
             String txCoordinatorId
     ) {
+        TxMeta txMeta = txStateStorage.get(txId);
+
+        // Check that a transaction has already been finished.
+        boolean transactionAlreadyFinished = txMeta != null && isFinalState(txMeta.txState());
+
+        // Check locksReleased flag. If it is already set, do nothing and return a successful result.
+        // Even if the outcome is different (the transaction was aborted, but we want to commit it),
+        // we return 'success' to be in alignment with common transaction handling.
+        if (transactionAlreadyFinished) {
+            if (txMeta.locksReleased()) {
+                return completedFuture(null);
+            }
+
+            assert !(txMeta.txState() == COMMITED && !commit) : "Not allowed to abort an already committed transaction.";
```

Comment:
```suggestion
            assert commit || txMeta.txState() == ABORTED : "Not allowed to abort an already committed transaction [txState=" + txMeta.txState() + "].";
```

Reply:
Original option seems more clear. And I'd also add an explanation why it's not valid to have abort request over already committed transaction, while it's fine to have e.g. commit over aborted.